File: backend/course-service/src/main/java/com/elearning/course/service/SectionService.java
package com.elearning.course.service;

import com.elearning.course.dto.request.SectionRequest;
import com.elearning.course.dto.response.SectionResponse;
import com.elearning.course.security.AuthenticatedUser;

public interface SectionService {
    SectionResponse create(Long courseId, SectionRequest request, AuthenticatedUser currentUser);
    SectionResponse update(Long sectionId, SectionRequest request, AuthenticatedUser currentUser);
    void delete(Long sectionId, AuthenticatedUser currentUser);
}

File: frontend/src/app/core/services/section.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable, inject } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { Section, SectionRequest } from '../models/course.model';

@Injectable({ providedIn: 'root' })
export class SectionService {
  private http = inject(HttpClient);
  private baseUrl = environment.courseApiUrl;

  create(courseId: number, request: SectionRequest): Observable<Section> {
    return this.http.post<Section>(`${this.baseUrl}/courses/${courseId}/sections`, request);
  }

  update(sectionId: number, request: SectionRequest): Observable<Section> {
    return this.http.put<Section>(`${this.baseUrl}/sections/${sectionId}`, request);
  }

  delete(sectionId: number): Observable<void> {
    return this.http.delete<void>(`${this.baseUrl}/sections/${sectionId}`);
  }
}
